vermouth/processors/water_bias.py

```diff
@@ -0,0 +1,157 @@
+    structure element in `water_bias` and assinged if
+    `auto_bias` is set to True. Using the `idr_regions`
+    argument the water_bias can be changed for
+    intrinsically disordered regions (IDRs). The IDR
+    bias superseeds the auto bias.
+
+    This Processor updates the system.gmx_topology_params
+    attribute.
+
+    ** Subclassing **
+    If the procedure by which to assign the water bias is
+    to be changed this processor is best subclassed and the
+    assign_residue_water_bias method overwritten.
+    """
+
+    def __init__(self,
+                 auto_bias,
+                 water_bias,
+                 idr_regions):
+        """
+
+
+        Parameters
+        ----------
+        auto_bias: bool
+            apply the automatic secondary structure
+            dependent water biasing
+        water_bias: dict[str, float]
+            a dict of secondary structure codes and
+            epsilon value for the water bias in kJ/mol
+        idr_regions:
+            regions defining the IDRs
+        prefix: str
+            prefix of the Go virtual-site atomtypes
+        system: :class:vermouth.System
+            the system of the molecules is used for
+            storing the nonbonded parameters
+        """
+        self.water_bias = water_bias
+        self.auto_bias = auto_bias
+        self.idr_regions = idr_regions
+        self.system = None
+
+    def assign_residue_water_bias(self, molecule, res_graph):
+        """
+        Assign the residue water bias for all residues
+        with a secondary structure element or that are
+        defined by the region selector. Region selectors
+        superceed the auto assignement.
+
+        Parameters
+        ----------
+        molecule: :class:vermouth.Molecule
+            the molecule
+        res_graph: :class:vermouth.Molecule
+            the residue graph of the molecule
+        """
+        for res_node in res_graph.nodes:
+            resid = res_graph.nodes[res_node]['resid']
+            chain = res_graph.nodes[res_node]['chain']
+            resname = res_graph.nodes[res_node]['resname']
+
+            if _in_region(resid, self.idr_regions):
+                eps = self.water_bias.get('idr', 0.0)
+            elif self.auto_bias:
+                sec_struc = res_graph.nodes[res_node]['cgsecstruct']
+                eps = self.water_bias.get(sec_struc, 0.0)
+            else:
+                continue
+
+            if eps == 0.0:
+                continue
+
+            vs_go_node = next(get_go_type_from_attributes(res_graph.nodes[res_node]['graph'],
+                                                          resid=resid,
+                                                          chain=chain,
+                                                          prefix=molecule.meta.get('moltype')))
+
+            # what is the blocks bb-type
+            bb_type = molecule.force_field.blocks[resname].nodes['BB']['atype']
```

Comment:
We'll regret having BB hardcoded here in a few years.

Reply:
OTOH, this is *extremely* model specific to begin with. 🤷

Reply:
I don't have a better idea either